keps/sig-api-machinery/5051-server-side-apply-unsetting-fields/README.md

```diff
@@ -0,0 +1,964 @@
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes an improvement to Server Side Apply to allow apply configurations to declare the
+intent that a field should not be set.
+
+## Motivation
+
+As of Kubernetes 1.32, Server side apply lacks a direct way to unset fields.
+
+If a field manager exclusively owns a field, the field manage may unset the field by applying a
+configuring that omits the field. But if the field manager does not exclusively own the field, there
+is no way for the field manager to unset the field using an apply request.
+
+Additionally, there is no way for a field manager to own a field that is unset.  This prevents a
+field manager from expressing the opinion that a field should be unset, and to force a conflict if
+another field manager attempts to set the field.
+
+### Goals
+
+- Introduce a way to express that a field should be unset in a Server Side Apply configuration.
+- Introduce field ownership of unset fields into the managedFields of Server Side Apply.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+Introduce a "marker" value that may be used in apply configurations to indicate that a value should
+be unset.
+
+The proposed marker value is: `{k8s_io__value: unset}`
```

Comment:
Are there any other k8s-features/SSA-features that use a marker value like {k8s_io__value: unset}?  Might be of use to show that there is some precedent for the format and the approach if so.

Reply:
There is not. We're diverging from the "all apply configurations are valid subsets of the full resource representation" here.

Reply:
Something to call out very thoroughly in docs, then.
- People have previously not need to know if their YAML file represents an incomplete object you can submit using a **create** verb or an apply configuration for SSA.
- We'll need to define which of these two meanings aligns with our specialist jargon term _manifest_

Reply:
Maybe we could accept `{k8s_io__value: unset}` for **create** / `PUT` too?

Reply:
Related, I think: if you have a manifest rather than an apply configuration, what `kubectl` subcommand do you use to ensure that the manifest represents your declared, desired state?

(I think I know the answer and I don't like it).